math/overflowHandler.go

```diff
@@ -0,0 +1,47 @@
+package math
+
+import (
+	"fmt"
+	builtinMath "math"
+)
+
+type overflowHandler struct {
+	err error
+}
+
+func NewOverflowHandler() *overflowHandler {
+	return &overflowHandler{}
+}
+
+// AddInt64 will add the two provided values. If an overflow occurs, the error will be stored internally
+func (handler *overflowHandler) AddInt64(a, b int64) int64 {
+	s := a + b
```

Comment:
Should return `builtinMath.MaxInt64` early if `handler.err != nil`. Same for `MulInt64`.